### PR TITLE
GG-24925: Added index.md to generate main page and added info to README.md

### PR DESCRIPTION
**Pull request by isapego**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,15 +18,104 @@ npm install -g @gridgain/thin-client
...
 npm link
 npm link @gridgain/thin-client #linking examples (optional)
 ```
+---------------------------------------------------------------------
 
-For more information, see [GridGain Node.JS Thin Client documentation](https://apacheignite.readme.io/docs/nodejs-thin-client).
\ No newline at end of file
+# Tests #
+
+NodeJS Client for GridGain contains [Jasmine](https://www.npmjs.com/package/jasmine) tests to check the behavior of the client. the tests include:
```
Comment (ptupitsyn) [inline README.md]: I think it is `Node.js`, not `NodeJS`

File: README.md [hunk 2]
```diff
@@ -18,15 +18,104 @@ npm install -g @gridgain/thin-client
...
+
+Call `npm test` command from `nodejs-thin-client` folder.
+
+### Run Examples Executors ###
+
+Call `npm run test:examples` command from `nodejs-thin-client` folder.
+
+### Run AuthTlsExample Executor ###
+
+It requires running GridGain server with non-default configuration (authentication and TLS switched on).
```
Comment (ptupitsyn) [inline README.md]: Language: `It requires running GridGain server` -> `GridGain server node is required`

File: README.md [hunk 3]
```diff
@@ -18,15 +18,104 @@ npm install -g @gridgain/thin-client
...
+
+  - The following files are needed:
+    - keystore.jks, truststore.jks - for the server side
+    - client.key, client.crt, ca.crt - for the client side
+
+2. Place client.key, client.crt and ca.crt files somewhere locally, eg. into the [examples/certs](./examples/certs) folder.
+
+3. If needed, modify `TLS_KEY_FILE_NAME`, `TLS_CERT_FILE_NAME` and `TLS_CA_FILE_NAME` constants in the example source file. The default values point to the files in the [examples/certs](./examples/certs) folder.
+
+4. Setup GridGain server to accept TLS - see appropriate GridGain documentation. Provide the obtained keystore.jks and truststore.jks certificates during the setup.
```
Comment (ptupitsyn) [inline README.md]: Link to related GG docs?

File: README.md [hunk 4]
```diff
@@ -18,15 +18,104 @@ npm install -g @gridgain/thin-client
...
+
+5. Switch on and setup authentication in GridGain server - see appropriate GridGain documentation.
```
Comment (ptupitsyn) [inline README.md]: Same as above - can we link the docs?

File: README.md [hunk 5]
```diff
@@ -18,15 +18,104 @@ npm install -g @gridgain/thin-client
...
+
+6. If needed, modify `USER_NAME` and `PASSWORD` constants in the example source file. The default values are the default GridGain username/password.
+
+## Additional Setup for FailoverExample ##
+
+1. Run three GridGain nodes. See appropriate GridGain documentation for more details.
```
Comment (ptupitsyn) [inline README.md]: Same as above

File: README.md [hunk 6]
```diff
@@ -18,15 +18,104 @@ npm install -g @gridgain/thin-client
...
+
+2. If needed, modify `ENDPOINT1`, `ENDPOINT2`, `ENDPOINT2` constants in an example source file - GridGain node endpoints.
+Default values are `localhost:10800`, `localhost:10801`, `localhost:10802` respectively.
+
+2. Run an example by calling `node FailoverExample.js`. 
+
+3. Shut down the node the client connected to (you can find it out from the client logs in the console).
```
Comment (ptupitsyn) [inline README.md]: `is connected to`

File: api_spec/index.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# NodeJS Client for GridGain
+
+This thin client allows your Node.JS applications to work with GridGain clusters via TCP.
```
Comment (ptupitsyn) [inline api_spec/index.md]: `Node.js`

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (vveider) [reply README.md]: Activated GridGain node?

**Review by ptupitsyn — APPROVED** [review]

**Pull request merged**